nomad/leader.go

```diff
@@ -1966,43 +1963,79 @@ func (s *Server) getOrCreateSchedulerConfig() *structs.SchedulerConfiguration {
 	return config
 }
 
+var minVersionKeyring = version.Must(version.NewVersion("1.4.0"))
+
 // initializeKeyring creates the first root key if the leader doesn't
 // already have one. The metadata will be replicated via raft and then
 // the followers will get the key material from their own key
 // replication.
-func (s *Server) initializeKeyring() error {
+func (s *Server) initializeKeyring(stopCh <-chan struct{}) {
+
+	logger := s.logger.Named("keyring")
 
 	store := s.fsm.State()
 	keyMeta, err := store.GetActiveRootKeyMeta(nil)
 	if err != nil {
-		return err
+		logger.Error("failed to get active key: %v", err)
+		return
 	}
 	if keyMeta != nil {
-		return nil
+		return
 	}
 
-	s.logger.Named("core").Trace("initializing keyring")
+	logger.Trace("verifying cluster is ready to initialize keyring")
+
+	versionCheck := func() bool {
+		members := s.serf.Members()
+		for _, member := range members {
+			build := member.Tags["build"]
+			memberVersion := version.Must(version.NewVersion(build))
+			if memberVersion.Core().LessThan(minVersionKeyring) {
+				return false
+			}
+		}
+		return true
+	}
```

Comment:
Could this use https://github.com/hashicorp/nomad/blob/v1.4.0/nomad/util.go#L146-L162?

I like the actual version checking better here, but that helper we normally use includes some safety/liveness checks that seem important. I think this approach breaks for folks who don't upgrade in place because the old rolled-out servers will still be in serf?

Reply:
Good catch. Will fix.

Reply:
Done. We could probably update that helper to use `Core()` as well, but that's future work 😀